### PR TITLE
Fix for mishandled " and & characters

### PR DESCRIPTION
**Pull request by DieterBuysAI**
This pull request fixes the bugs described in issues #14 and #16

### CODE REVIEW

File: secretary.py [hunk 1]
```diff
@@ -306,20 +306,30 @@ def _prepare_template_tags(self, xml_document):
...
-    def _unescape_entities(self, xml_text):
-        # unescape XML entities gt and lt
+    @staticmethod
+    def _unescape_entities(xml_text):
+        """
+        Strips tags of the form <text:span ...> from inside Jinja elements
+        and unescapes HTML codes for >, <, & and "
+        """
         unescape_rules = {
+            r'(?is)({[{|%].*)(<.?text:s.*?>)(.*[%|}]})': r'\1\3',
```
Comment (DieterBuysAI) [inline secretary.py]: This is now the first rule, which means that the `<text:s...>` tags will get ripped out before the `< > & "` get replaced. In addition I've changed the matching rule to `<.?text:s.*?>` which will non-greedily go to the first `>` that closes the tag, whereas the rule previously was only doing `<.?text:s.?>` which would only match one character after the s.

File: secretary.py [hunk 2]
```diff
@@ -306,20 +306,30 @@ def _prepare_template_tags(self, xml_document):
...
         for p, r in unescape_rules.items():
-            xml_text = re.sub(p, r, xml_text)
+            count = None
+            while count != 0:
+                xml_text, count = re.subn(p, r, xml_text)
 
         return xml_text
 
-    def _encode_escape_chars(self, xml_text):
+    @staticmethod
```
Comment (DieterBuysAI) [inline secretary.py]: I figured these two could be made into static methods since they do not rely on `self`.

File: secretary.py [hunk 3]
```diff
@@ -306,20 +306,30 @@ def _prepare_template_tags(self, xml_document):
...
-            r'(?is)({[{|%].*)(<.?text:s.?>)(.*[%|}]})': r'\1 \3',
+            r'(?is)({[{|%].*)(&amp;)(.*[%|}]})': r'\1&\3',
+            r'(?is)({[{|%].*)(&quot;)(.*[%|}]})': r'\1"\3'
         }
 
         for p, r in unescape_rules.items():
-            xml_text = re.sub(p, r, xml_text)
+            count = None
+            while count != 0:
+                xml_text, count = re.subn(p, r, xml_text)
```
Comment (DieterBuysAI) [inline secretary.py]: Using the while loop ensures all the special characters get converted which allows us to deal with expressions that may have multiple comparisons and pairs of double quotes in them such as `{% if x > 0 and x < 10 and y != "Foo" %}`.

File: secretary.py (hunk 1 above)
Comment (christopher-ramirez) [inline secretary.py]: Actually `text:s.?` will match more than one character. A non-greedily match will only match the first char after **`s`**. So if you need to escape `text:space` using `r'text:s.*?'` you will be only matching **`text:sp`**.

This the reason why your version is not passing automated tests.

**Pull request merged**